tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1482 @@
+        bodyContent = JsObject(bodyContent.fields + ("annotations" -> annos))
+      }
+    }
+
+    val respFuture =
+      if (update) createEntity(path, bodyContent.toString, Map("overwrite" -> "true"))
+      else createEntity(path, bodyContent.toString)
+    val resp = respFuture.futureValue
+    val r = new RestResult(resp.status, getRespData(resp))
+    if (expectedExitCode != DONTCARE_EXIT) {
+      if (r.statusCode.intValue != expectedExitCode)
+        r.statusCode.intValue shouldBe expectedExitCode
+    }
+    r
+  }
+
+  /**
+   * Binds package. Parameters mirror those available in the REST.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def bind(provider: String,
+                    name: String,
+                    parameters: Map[String, JsValue] = Map(),
+                    annotations: Map[String, JsValue] = Map(),
+                    expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+    val params = convertMapIntoKeyValue(parameters)
+    val annos = convertMapIntoKeyValue(annotations)
+
+    val (ns, packageName) = this.getNamespaceActionName(provider)
+    val path = Path(s"$basePath/namespaces/${wp.namespace}/$noun/$name")
+    val binding = JsObject("namespace" -> ns.toJson, "name" -> packageName.toJson)
+    val bodyContent = JsObject("binding" -> binding.toJson, "parameters" -> params, "annotations" -> annos)
+    val respFuture = createEntity(path, bodyContent.toString, Map("overwrite" -> "false"))
+    val resp = respFuture.futureValue
+    val r = new RestResult(resp.status, getRespData(resp))
+    if (expectedExitCode != DONTCARE_EXIT) {
+      if (r.statusCode.intValue != expectedExitCode)
+        r.statusCode.intValue shouldBe expectedExitCode
+    }
+    r
+  }
+
+}
+
+class WskRestApi extends RunWskRestCmd with BaseApi {
+  protected val noun = "apis"
+
+  /**
+   * Creates and API endpoint. Parameters mirror those available in the REST.
+   *
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def create(basepath: Option[String] = None,
+                      relpath: Option[String] = None,
+                      operation: Option[String] = None,
+                      action: Option[String] = None,
+                      apiname: Option[String] = None,
+                      swagger: Option[String] = None,
+                      responsetype: Option[String] = None,
+                      expectedExitCode: Int = SUCCESS_EXIT,
+                      cliCfgFile: Option[String] = None)(implicit wp: WskProps): RestResult = {
+    val (ns, actionName) = this.getNamespaceActionName(action map { a =>
+      a
+    } getOrElse "")
```

Comment:
What happens if action name is an empty string?

Reply:
We will return notfound error in the response.